servlet/src/main/java/io/undertow/servlet/spec/HttpServletRequestImpl.java

```diff
@@ -255,10 +255,12 @@ public HttpServletMapping getHttpServletMapping() {
                 }
                 break;
             case EXTENSION:
-                matchValue = match.getMatched().substring(0, match.getMatched().length() - match.getMatchString().length() + 1);
-                if(matchValue.startsWith("/")) {
-                    matchValue = matchValue.substring(1);
-                }
+                String remaining = match.getRemaining();
+                String matched = remaining != null ? remaining : match.getMatched();
```

Comment:
This line fixes our bug.

Previously `match.getMatched()` returned the matched prefix, even in the case of an extension match, if both types of matches occurred. Now we use remaining if present, falling back to matched if remaining is null.
If we have a remaining segment, that's where we can find the matched extension. If there's no remaining segment, the previous implementation was correct.

Reply:
Updated, this comment referred to a change that avoided throwing exceptions, but did not fix the underlying issue diverging from the servlet spec.